Extensions/TweenBehavior/tweenruntimebehavior.ts

```diff
@@ -632,39 +654,192 @@ namespace gdjs {
+        durationValue
+      );
+      this._setupTweenEnding(identifier, destroyObjectWhenFinished);
+    }
+
+
+
+    /**
+     * Add an object color HSL tween, with the "to" color given using HSL (H: any number, S and L: 0-100).
+     * @param identifier Unique id to idenfify the tween
+     * @param toHue The target hue, or the same as the from color's hue if blank
+     * @param animateHue, include hue in calculations, as can't set this to -1 as default to ignore
+     * @param toSaturation The target saturation, or the same as the from color's saturation if blank
+     * @param toHue The target lightness, or the same as the from color's lightness if blank
+     * @param easingValue Type of easing
+     * @param durationValue Duration in milliseconds
+     * @param destroyObjectWhenFinished Destroy this object when the tween ends
+     */
+    
+    addObjectColorHSLTween(
+      identifier: string,
+      toHue: string,
+      animateHue: string,
+      toSaturation: string,
+      toLightness: string,
+      easingValue: string,
+      durationValue: float,
+      destroyObjectWhenFinished: boolean,
+    ) {
+      const that = this;
+      
+      let fromColorAsHSL : number[] = [], toColorAsHSL : number[] = [];
+      if (!this._isActive) {
+        return;
+      }
+      // @ts-ignore - objects are duck typed
+      if (!this.owner.getColor || !this.owner.setColor) {
+        return;
+      }
+      if (!!TweenRuntimeBehavior.easings[easingValue]) {
+        return;
+      }
+      
+      if (this._tweenExists(identifier)) {
+        this.removeTween(identifier);
+      }
+      
+      // @ts-ignore - objects are duck typed
+      let fromColor: any[] = this.owner.getColor().split(';');
+      fromColor = fromColor.map((x: Number) => Number(x) / 255);
+      let v = Math.max(fromColor[0], fromColor[1], fromColor[2]), c = v - Math.min(fromColor[0], fromColor[1], fromColor[2]), f = (1 - Math.abs(v + v - c - 1));
+      let h = c && ((v === fromColor[0]) ? (fromColor[1] - fromColor[2]) / c: ((v === fromColor[1]) ? 2 + (fromColor[2] - fromColor[0]) / c: 4 + (fromColor[0] - fromColor[1]) / c)); 
+      fromColorAsHSL = [Math.round(60 * (h < 0 ? h + 6: h)), Math.round((f ? c / f: 0) * 100), Math.round(((v + v - c) / 2) * 100)];
+
+      let toH: number = 0, toS: number = 0, toL: number = 0;
+      if(animateHue === "no" || toHue.trim() === "") {
+        toH = fromColorAsHSL[0];
+      }
+      else if(!Number.isNaN(Number.parseFloat(toHue))) {
+        toH = parseFloat(toHue);
+      } else {
+        return;
+      }
+      
+      if("-1" === toSaturation.trim() || "" === toSaturation.trim()) {
+        toS = fromColorAsHSL[1];
+      } else if(!Number.isNaN(Number.parseFloat(toSaturation))) {
+        toS = parseFloat(toSaturation);
+        toS = Math.min(Math.max(toS, 0), 100); // Clamp
+      } else {
+        return;
+      }
+
+      if("-1" === toLightness.trim() || "" === toLightness.trim()) {
```

Comment:
Same, check with a debugger or console.log if you want to be sure.

Reply:
Also tidied. How do I see the console window of the running game preview? - I can see it for the Electron app, and can use alert() within the code, but I can't turn on the Chrome Dev tools for the game itself. Is there a magic keyboard shortcut?